CPAC/pipeline/utils.py

```
@@ -17,13 +17,143 @@
+    command = ["3drefit", "-TR", str(new_pixdim4), file_path]
+
+    try:
+        subprocess.run(command, check=True)
+        IFLOGGER.info(f"Successfully updated TR to {new_pixdim4} seconds.")
+    except subprocess.CalledProcessError as e:
+        error_message = f"Error occurred while updating the file: {e}"
+        raise subprocess.CalledProcessError(error_message)
+
+
+def validate_outputs(input_bold, RawSource_bold):
+    """Match pixdim4/TR of the input_bold with RawSource_bold.
+
+    Parameters
+    ----------
+    input_bold : str
+        Path to the input BOLD file.
+    RawSource_bold : str
+        Path to the RawSource BOLD file.
+
+    Returns
+    -------
+    output_bold : str
+        Path to the output BOLD file.
+
+    Raises
+    ------
+    Exception
+        If there is an error in finding or updating pixdim4.
+    """
+    try:
+        output_bold = input_bold
+        output_pixdim4 = find_pixdim4(output_bold)
+        source_pixdim4 = find_pixdim4(RawSource_bold)
+
+        if output_pixdim4 != source_pixdim4:
+            IFLOGGER.info(
+                "TR mismatch detected between output_bold and RawSource_bold."
+            )
+            IFLOGGER.info(f"output_bold TR: {output_pixdim4} seconds")
+            IFLOGGER.info(f"RawSource_bold TR: {source_pixdim4} seconds")
+            IFLOGGER.info(
+                "Attempting to update the TR of output_bold to match RawSource_bold."
+            )
+            update_pixdim4(output_bold, source_pixdim4)
+        else:
+            IFLOGGER.info("TR match detected between output_bold and RawSource_bold.")
+            IFLOGGER.info(f"output_bold TR: {output_pixdim4} seconds")
+            IFLOGGER.info(f"RawSource_bold TR: {source_pixdim4} seconds")
+        return output_bold
+    except Exception as e:
+        error_message = f"Error in validating outputs: {e}"
+        IFLOGGER.error(error_message)
+        return output_bold
```

Do we want to raise an exception here instead of (log an error message and return the path to the bold file)?

---

Good catch, I am not sure if we want to `totally crash` (by raising exception)  CPAC here or if we only say following resources were not validated but still its here?

---

or probably add a swtich to the config for raising exception or continuing?

---

@sgiavasis , mentioned this being such a small add-on, it is very unlikely to fail and if fails we can continue instead of crashing the CPAC.